#,Status,Project,Tracker,Priority,Subject,Assigned to,Category,Target version,Author,Start,Due date,% Done,Estimated time,Created,Updated,Likelihood,Severity,Found in Version,Impact,Expected Use,OS,Support Group,Description
2629,Resolved,VisIt,Bug,Normal,FindVisItQt5.cmake with Darwin,Kevin Griffin,"",2.13.0,Allen Sanderson,06/07/2016,"",100,"",06/07/2016 05:55 pm,05/17/2017 06:33 pm,5 - Always,3 - Major Irritation,trunk,"","",All,Any,"With Darwin this call fails:

      THIRD_PARTY_INSTALL_LIBRARY(${qtlib_location})

because the libraries do not have an extension (.a .la).

For Qt5 on Darwin - the libraries are frameworks now. But we are still treating them as if they were a dylib. For now I have added a hack in FindVisItQt5.cmake for Darwin

      if (APPLE)
          string(CONCAT qtlib_location ${qtlib_location} "".la"")                 
      endif()      

to add a dummy extension. This seems to get past the problem BUT we need to treat Qt5 on darwin as a framework rather than libraries. This comment also applies to FindQwt.cmake"
2491,Resolved,VisIt,Bug,Normal,Upgrade to Python 2.7.11,Kathleen Biagas,"",2.11.0,Allen Sanderson,12/29/2015,"",100,"",12/29/2015 05:50 pm,06/28/2016 03:38 pm,3 - Occasional,2 - Minor Irritation,trunk,"","",All,Any,I am suggesting we upgrade to Python 2.7.11. I found that on OS X 10.11 that python's internal struct did not build correctly. I used 2.7.11 and it built fine.
2023,Resolved,VisIt,Bug,High,The makemoviemain script could not locate the makemovie script!,Eric Brugger,"",2.8.2,Allen Sanderson,10/20/2014,"",0,"",10/20/2014 01:50 pm,11/25/2014 11:51 am,5 - Always,3 - Major Irritation,2.8.0,"","",All,Any,"When making movies from the command line VisIt fails:

The makemoviemain script could not locate the makemovie script!

To reproduce the bug make a movie like using this command:

visit  -movie -start 0 -end 2 -frame 0 -geometry 1024x1280 -format tiff -output movie -sessionfile test.session"
1521,Resolved,VisIt,Bug,Normal,Switching active source does not fully update the variable list,Kathleen Biagas,"",2.7,Allen Sanderson,06/28/2013,"",0,4.0,06/28/2013 12:19 pm,07/03/2013 02:46 pm,4 - Common,3 - Major Irritation,trunk,"","",All,Any,"If two databases are loaded when switching from the second back to the first it is not possible to create a plot. The viewer menus get updated with the correct variable list but under the hood VisIt retains the other database's variable list. Thus if the variable names are different between two databases those that are different are not accessible.

Open this file:

data/visit_dist_data/ucd3d.silo

Open this file:

data/visit_dist_data/multi_ucd3d.silo

Create a simple pseducolor plot of this variable: vec_magnitude

Now set the active source to

data/visit_dist_data/ucd3d.silo

Create a simple pseducolor plot of this variable: vel_magnitude

While the viewer menus are updated VisIt still seems to think the other database is active and gives:

An invalid variable (vel_magnitude) was specified.


So somewhere the database information is not being fully updated.
"
360,Resolved,VisIt,Feature,Normal,New windows need to inherit opened databases,Brad Whitlock,"",2.2,Allen Sanderson,08/25/2010,"",0,"",08/25/2010 01:00 pm,01/14/2011 04:18 am,"","","",3 - Medium,4 - Common,All,Any,"When opening a new window the user needs to be given the option of inheriting the ""Selected Files"" from the current window. The current default is that the window does not have any open databases which is annoying. This option would allow the window to inherit the open databases without having to ""Clone"" the window and then delete the plots which is the current work around.
"
2024,Pending,VisIt,Bug,Normal,command line engine args creates extra (bad) args,"","",3.0,Allen Sanderson,10/20/2014,"",0,"",10/20/2014 01:53 pm,06/30/2015 07:15 pm,3 - Occasional,3 - Major Irritation,2.7.3,"","",All,Any,"There is a long standing bug if one sends -np on the command line for engine it gets parsed but then adds a bunch of other engine args including defaulting to mpirun when it should use what is in the session file.

BAD:  -engineargs -l;mpirun;-np;64;-nn;8;-b;smithp-ash-cs -launchengine localhost  <<< SHOULD NOT BE mpirun but qsub/mpirun

Running: cli2.8.1 -start 0 -end 2 -frame 0 -geometry 1024x1280 -format tiff -output movie -sessionfile ../BSFO2.session -nowin -s /uufs/ash.peaks/sys/pkg/VisIt/Install/gcc/2.8.1/linux-x86_64/bin/makemoviemain.py -engineargs -l;mpirun;-np;64;-nn;8;-b;smithp-ash-cs -launchengine localhost

GOOD: -engineargs -b;smithp-ash-cs

Running: cli2.8.1 -start 0 -end 2 -frame 0 -geometry 1024x1280 -format tiff -output movie -sessionfile ../BSFO2.session -nowin -s /uufs/ash.peaks/sys/pkg/VisIt/Install/gcc/2.8.1/linux-x86_64/bin/makemoviemain.py -engineargs -b;smithp-ash-cs

If I add a -l qsub/mpirun when the script works 

Running: cli2.8.1 -start 0 -end 2 -frame 0 -geometry 1024x1280 -format tiff -output movie -sessionfile ../BSFO2.session -nowin -s /uufs/ash.peaks/sys/pkg/VisIt/Install/gcc/2.8.1/linux-x86_64/bin/makemoviemain.py -engineargs -l;qsub/mpirun;-np;64;-nn;8;-b;smithp-ash-cs -launchengine localhost"
2720,Pending,VisIt,Bug,High,Colormap legend is not correct for current plot and sometimes for original data,Kathleen Biagas,"",2.13.3,Allen Sanderson,11/23/2016,"",0,"",11/23/2016 11:59 am,05/10/2018 08:36 pm,3 - Occasional,4 - Crash / Wrong Results,trunk,"","",All,Any,"I have a two level data set. Here are the result of min/max from the query.

Original Data:
sigmaT4/0 -- Min = 0 (zone 0 in patch level0, patch0 at coord <7.2352, -0.274519, -0.259955>)
sigmaT4/0 -- Max = 156318 (zone 1523 in patch level1, patch2553 at coord <28.0883, 14.4809, 2.79451>)

Current Plot L0 only
sigmaT4/0 -- Min = 0 (zone 0 in patch level0, patch0 at coord <7.2352, -0.274519, -0.259955>)
sigmaT4/0 -- Max = 37428 (zone 196 in patch level0, patch0 at coord <7.7648, 0.274519, 0.259955>)

Current Plot L1 only
sigmaT4/0 -- Min = 233.22 (zone 624 in patch level1, patch1981 at coord <23.4543, -0.0686296, 2.14462>)
sigmaT4/0 â€” Max = 156318 (zone 1523 in patch level1, patch2553 at coord <28.0883, 14.4809, 2.79451>)

Plot 1 contains both L0/L1 and uses the original data the color range is as I would expect: 0-156318
(not shown) 
Plot 1a contains both L0/L1 and uses the current data the color range MAX is NOT as I would expect: 156318 (22160 wrong)

Plot 2 contains L0 and uses the current data the color range MAX is NOT as I would expect: 37428 (22160 wrong)
Plot 3 contains L1 and uses the current data the color range MAX is NOT as I would expect: 156318 (22160 wrong)

The legend MIN for both plot 1a, 2 and 3 are CORRECT 0, 0, and 233.22 respectively.

FWIW I have verified via a simple PS plot of rendering a single field sans any operators and swapping between original and current plot changes the legend. It should not. 


I also saw some other issues when restoring a session. Here the Min/Max in the legend were wrong. I also a similar issue when I did a replace operation. I was not able to even figure out the origin of those values.



"
358,Resolved,VisIt,Bug,Normal,plugin manager database settings not saved,Eric Brugger,"","",Allen Sanderson,08/25/2010,"",0,"",08/25/2010 12:46 pm,12/11/2013 02:30 pm,5 - Always,3 - Major Irritation,2.0.1,"","",OSX,Any,"remove ~/.visit/comnfig

start visit

open plugin manager

all databases are selected. 

deselect all

select the ones I want, add them to the referred list

apply

save

quit visit

restart visit

open plugin manager
the preferred list is correct

all of the databases are selected again
"
359,Pending,VisIt,Bug,Normal,restoring session files get confused on which database to use,"","","",Allen Sanderson,08/25/2010,"",0,"",08/25/2010 12:55 pm,09/22/2010 02:17 pm,3 - Occasional,2 - Minor Irritation,2.0.1,"","",All,Any,"I restore a session file that uses silo reader, works fine. Restore another session file that uses M3DC1 reader. MDS server tries H5Part first which throws HDF5 errors."
362,Developer Review,VisIt,Bug,Normal,pseudocolor plot with 1D rectilinear grids fails,"","","",Allen Sanderson,08/25/2010,"",0,"",08/25/2010 04:57 pm,09/22/2010 01:39 pm,3 - Occasional,2 - Minor Irritation,2.0.1,"","",All,Any,"I have a 1D rectilinear grid of data 100x1x1. I was expecting a line from pseudocolor plot but got nothing. Even asking for rendering of points got nothing. But the legend was correct as was the bounding box.

This mesh renders fine in mesh plot - I see a line.

My guess is that pseudocolor plot is trying to make triangles but there are no triangles to make so it does nothing. It would seem to me that if it can not generate tris then it should try lines. If no lines then points. If no points then an error message saying no geometry.

I can change my data to an unstructured grid and it works. But I really have a rectilinear 1d grid."
367,Developer Review,VisIt,Feature,Normal,Volume plot opacity pixmap,"","","",Allen Sanderson,08/27/2010,"",0,"",08/27/2010 02:18 pm,09/14/2010 04:42 pm,"","","",3 - Medium,3 - Occasional,All,Any,"Volume plot opacity pixmap is backwards when showing colors. If the opacity is 100% the color is white. It would seem that it should be the color. I would like to suggest that it be flipped for both when showing the color and when turned off (no color).

Of course the preference is to use the same color table editor as used in the color table GUI."
369,Resolved,VisIt,Bug,Normal,Discrete Colormaps showing as continuous in the legend,Allen Sanderson,"","",Allen Sanderson,08/30/2010,"",0,"",08/30/2010 04:20 pm,12/11/2013 02:18 pm,3 - Occasional,2 - Minor Irritation,2.0.1,"","",All,Any,As the title says I have a discrete colormap with two colors but it shows up as a continuous map in the legend. 
409,Developer Review,VisIt,Bug,Normal,pseduocolor plot - point cloud scaling by variable,"","","",Allen Sanderson,09/30/2010,"",0,"",09/30/2010 03:58 pm,11/02/2010 03:00 pm,3 - Occasional,4 - Crash / Wrong Results,trunk,"","",All,Any,"I have a pseudo color plot that is plotting a point cloud of data. If I select the point type to be icosahedron and scale by the default variable. Sometimes I get the correct result. Sometimes I get nothing. When I get the correct result and go to another time step I typically get nothing. If I go back to the time step where the result was correct previously I get nothing. As such, it seems like the scaling feature has some problems. Oddly with another data set it seems to be working okay.

I can easily supply a test data set and session file."
478,Pending,VisIt,Bug,Normal,MDS Server - restoring session reader errors.,Jeremy Meredith,"","",Allen Sanderson,11/16/2010,"",0,"",11/16/2010 01:44 pm,12/07/2010 12:29 pm,3 - Occasional,3 - Major Irritation,trunk,"","",All,Any,"I have a session file with three files with specific readers specified, visit can not open one of the three but then continues to try with other readers.


The session file xml:
              <Field name=""localhost:/Projects/Data/Fusion/NIMROD/87009/phi_BT.h5"" type=""string"">""H5Nimrod_1.0""</Field>
              <Field name=""localhost:/Projects/Data/Fusion/Siesta/siesta_cdx.silo.h5"" type=""string"">""Silo_1.0""</Field>
              <Field name=""localhost:/Projects/latex_it"" type=""string"">""MDSplus_1.0""</Field>

The last file being read is going to throw InvalidFilesException - which is correct.

VisIt imported a session from: /Projects/VisIt/trunk/src/LineSampler.session.
Shortly thereafter, the following occured...
VisIt could not read from the file ""/Projects/latex_it"".
The generated error message was:
There was an error opening /Projects/latex_it. It may be an invalid file. VisIt tried using the following file format readers to open the file: MDSplus


However, then for unknown reasons VisIt tries to open the file using the H5Nimrod reader not once but twice.

First errant reading:

Shortly thereafter, the following occured...
VisIt cannot read the SIL for the file ""/Projects/latex_it"" on host localhost.
The metadata server returned the following message:
There was an error opening /Projects/latex_it. It may be an invalid file.
VisIt tried using the following file format readers to open the file: H5Nimrod
If you know the specific format reader VisIt should use to
read this data, you can use Open As... (GUI) or
'-o <file>,<plugin> (CL arg.) and identify that specific reader
for VisIt to try. This will possibly give more information on
the exact error.

Second Errant reading:

Shortly thereafter, the following occured...
VisIt could not read from the file ""/Projects/latex_it"".
The generated error message was:
There was an error opening /Projects/latex_it. It may be an invalid file.
VisIt tried using the following file format readers to open the file: H5Nimrod
If you know the specific format reader VisIt should use to
read this data, you can use Open As... (GUI) or
'-o <file>,<plugin> (CL arg.) and identify that specific reader
for VisIt to try. This will possibly give more information on
the exact error.

Then, after trying to examine the Active source list, it tries all of the readers in the preferred list twice. (Error below)

VisIt could not read from the file ""/Projects/latex_it"".
The generated error message was:
There was an error opening /Projects/latex_it. It may be an invalid file.
VisIt tried using the following file format readers to open the file: GTC, H5Nimrod, H5Part, M3D, M3DC1, Vs
If you know the specific format reader VisIt should use to
read this data, you can use Open As... (GUI) or
'-o <file>,<plugin> (CL arg.) and identify that specific reader
for VisIt to try. This will possibly give more information on
the exact error.

Shortly thereafter, the following occured...
VisIt cannot read the SIL for the file ""/Projects/latex_it"" on host localhost.
The metadata server returned the following message:
There was an error opening /Projects/latex_it. It may be an invalid file.
VisIt tried using the following file format readers to open the file: GTC, H5Nimrod, H5Part, M3D, M3DC1, Vs
If you know the specific format reader VisIt should use to
read this data, you can use Open As... (GUI) or
'-o <file>,<plugin> (CL arg.) and identify that specific reader
for VisIt to try. This will possibly give more information on
the exact error.


So it would seem that if a reader is specified in a restored session file and it fails then VisIt should quit rather than trying other readers?????

Next, it seems that the active source list is screwed up because of the above error. The active source list shows one file: siesta_cdx.silo.h5 which would appear to have been opened correctly because multiple plots are drawn based on it.

However, ""File Info"" say there are no open files. Further, ""Plots Add"" is deactivated. So at least there is consistency in VisIt being wrong.

It is possible to reopen the file siesta_cdx.silo.h5 and have the ""File Info"" correct and the ""Plots Add"" active.

Finally, visit continues to try and open the file and reports the error - it needs to quit.
"
499,Pending,VisIt,Bug,Normal,not all sources are saved in the session file,"","","",Allen Sanderson,12/03/2010,"",0,"",12/03/2010 12:32 pm,12/14/2010 04:49 pm,3 - Occasional,2 - Minor Irritation,2.0.1,"","",All,Any,"Open any two files, save the session file, restore the session file and only one file is in the ""Active Source"" list

My examples:

open MDSplus file - ""NSTALL_NOTES""
open silo file - siesta_cdx.silo.h5

save session

session file contents:

                <Field name=""SOURCE00"" type=""string"">localhost:/Users/
allen/Projects/Data/Fusion/Siesta/siesta_cdx.silo.h5</Field>
            </Object>
            <Object name=""SourcePlugins"">
                <Field name=""localhost:/Projects/VisIt/trunk/src/
A.cli.1.vlog"" type=""string"">MDSplus_1.0</Field>
                <Field name=""localhost:/Projects/VisIt/trunk/src/
INSTALL_NOTES"" type=""string"">MDSplus_1.0</Field>
                <Field name=""localhost:/Users/allen/Projects/Data/
Fusion/Siesta/siesta_cdx.silo.h5"" type=""string"">Silo_1.0</Field>


Restore session ... only siesta data shows up in the ""Active Source""
Whereas I would expect to see both.

Session file needs to save all sources whether used or not. Or active or not. Especially given that the plugins used is correct.
"
545,Pending,VisIt,Bug,Normal,Logicals in expressions can not be returned direclty,"","","",Allen Sanderson,01/07/2011,"",0,"",01/07/2011 03:45 pm,01/11/2012 05:18 pm,5 - Always,4 - Crash / Wrong Results,trunk,"","",All,Any,"For an expression I found that this statement while not returning an error gives bogus data of I believe all zero

ne( trapped, pos_cmfe(<[1]id:trapped>, particles, 0) )

Instead I must do the following:

if( ne( trapped, pos_cmfe(<[1]id:trapped>, particles, 0) ), 1, 0)


From Hank: The logicals return Booleans and that is probably tripping things up down stream.  We probably need code to detect when Booleans are being sent out of the EEF and translate to floats....

If it can not be done then we need an error in the expression parsing 

From Cyrus:
Our Pseudocolor plot handles vtkIntArrays just fine, but not vtkUnsignedCharArrays.

After the vtk upgrade, we will want to support native precision pipelines (not a small undertaking)
In light of this should we just support, directly plotting vars represented by vtkUnsignedChar?

"
679,Pending,VisIt,Feature,Normal,Addition of logical bounds to Silo Reader,"","","",Allen Sanderson,04/07/2011,"",0,"",04/07/2011 02:46 am,01/11/2012 05:21 pm,3 - Occasional,1 - Not Serious,trunk,3 - Medium,2 - Rare,All,Any,"I was wondering if would be possible to add in the setting of logical bounds in the meta data for silo files that contain structured data. I have some silo data that I would like have the logical bounds presentable to the user. I see places where rectilinear meshes are called out but I just do not have the knowledge to muck about - and given the number of people using the reader it is probably best that I do not. It is fairly easy once you know the logical bounds.

The main call

avtMeshMetaData::SetBounds()

or

replace the first argument (NULL) when calling

avtMeshMetaData *mmd = new avtMeshMetaData(NULL,
                                                     extents_to_use, ... );

in the file reader.

It is not critical but I thought it might be nice to have the bounds feature in some other readers.



Response from Mark:


Hi Allen,

Can you add a visit issue ticket for this request?

I looked at the reader briefly and it does serve extents up to VisIt if they are available in the quadmesh objects it reads.

For single block, Silo's DBPutQuadmesh call will compute and store extents in the file for you. 

For multi-block, the DBPutMultimesh call requires that you use the DBOPT_EXTENTS_SIZE and DBOPT_EXTENTS DBoptlist options to output extents for each block. If you do not do this, there is no other way for VisIt to read them during PopulateMD. That is because the Silo plugin will NOT attempt to read each quadmesh block of a multi-block object solely to gather up extents information.

Mark
"
933,Pending,VisIt,Bug,Normal,Silo oddities with cycles,"","","",Allen Sanderson,01/10/2012,"",0,"",01/10/2012 04:14 pm,02/02/2012 07:32 pm,3 - Occasional,2 - Minor Irritation,trunk,"","",All,Any,"Finally submitting this bug. Summary: cycles are not always read/displayed correctly

ARS:I had several oddities with a series of silo files from a fusion app scientist at MIT, Linda Sugiyama. All of the time slice files are in individual silo files. 

ARS:Case 1: Try harder is off,  file info shows first time but the cycles are present,

MM: I think thats correct behavior if your filenames have cycle numbers in
them. It can 'guess' cycles from the filenames. But, it gets only the
currently open file's time correct 'cause it actually reads that from
the file.

Case 2: Try harder is on,  file info shows all the times but the cycles are all zero.

MM:Sounds like incorrect behavior if the filenames have cycle numbers in
them. Should get all cycles and all times.


ARS: The other part is that when we did a plot using TimeLoopFilter (via
LineSampler) we found that all the times were correct except for the
current time slice, we tried the same using the cycles for axis and
found the same the cycle of the current time slice was wrong. Only the
time slice always correct.

MM:In this case, it is going to open all the files, right? Whenever it
opens a file, it can at least get correct time/cycle for that file
'cause its not any real additional cost to go read it.




From Linda Sugiyama:

Regarding cycle/time series cases 1 and 2:

The 'current time' existed because an initial, separate window was opened
with this time and was still displaying data from it,
while the time series was being done.
(It appears to be necessary to open one file and plot some data initially
in order to for VisIt to be able to access the file series --
true, or avoidable?).

The 'time' series list used the cycle numbers from the file names
for all except the 'current' file.  There it used the actual time.
It should use a consistent choice.

In Case 1, an additional bug is that
the cycle number of the 'current' file is always set to zero in the cycle
listing. The position of the file in both time and cycle lists is
correct, based on the cycle number in the file name,
but now the cycle list has a zero-number cycle out of sequence.
(In fact, there are two files with cycle number zero, since the first time in the series has file/cycle number zero.)

Initially opening the first file of the series still gives an
'out-of-sequence' error warning even though the numbers are consistent.
This file had file/cycle number zero and its actual time (note t>0) was much smaller than the cycle-number increment between consecutive files, so nominally both time and cycle lists are OK.  Allen's time-series operator makes an acceptable 'time' plot, even including the first file.
(Not thoroughly tested - I think selecting 'Time series' works, but 'cycle number' puts negative cycle numbers on the X axis, centered around zero.)
"
1906,Pending,VisIt,Bug,Normal,Recentering of zonal AMR data does not fully work,"","","",Allen Sanderson,07/09/2014,"",0,"",07/09/2014 04:18 pm,08/26/2014 04:32 pm,3 - Occasional,4 - Crash / Wrong Results,trunk,"","",All,Any,"I have an AMR data set with a single level. The data is zonal. When re-centered to be nodal via the expression operator the results appear to the be correct. However, if one then does an iso-surface operation the results are wrong. 

I exported this same data set an as a VTK dataset and re-read it in and was able to successfully iso-surface the data. 

As such, I believe that there is an issue with re-centering and AMR data."
1909,Resolved,VisIt,Bug,Low,Transform operator - avtCoordSystemConvert - FixWraparounds,"","","",Allen Sanderson,07/10/2014,"",0,"",07/10/2014 02:50 pm,07/18/2014 02:32 pm,5 - Always,1 - Not Serious,trunk,"","",All,Any,"It would appear that FixWraparounds located in  avtCoordSystemConvert.C does a conversion of VTK_POLY_DATA to a VTK_UNSTRUCTURED_GRID. Which is not correct in two ways. The conversion does not keep any of the lines. So all geometry is lost.  Points are passed through. But further transform should not be returning VTK_UNSTRUCTURED_GRID but another VTK_POLY_DATA.

This problem occurs in the case of integral curves operator that returns a VTK_POLY_DATA structure that contains only lines. 

I am not quite sure that should happen as for now I have a check that by-passes this conversion if a VTK_POLY_DATA is encountered that contains only line.

This can be recreated by creating integral curves and doing a conversion from cartesian to cylindrical coordinates (would deed to remove my check)."
2207,Pending,VisIt,Bug,Normal,netCDF reader and auto domain decomposition ,"","","",Allen Sanderson,04/03/2015,"",0,"",04/03/2015 03:29 pm,04/07/2015 06:53 pm,4 - Common,4 - Crash / Wrong Results,2.9.0,"","",All,Any,"When reading a netcdf with the basic reader in parallel the domain decomposition works but only one domain gets rendered.

This occurs with Pseudocolor and vector plots. It also prevents integral curves from being advected."
2518,Pending,VisIt,Bug,Low,svn_bin/visit-install,"","","",Allen Sanderson,01/27/2016,"",0,"",01/27/2016 05:02 pm,02/23/2016 06:57 pm,3 - Occasional,1 - Not Serious,2.10.0,"","",All,Any,"We need to make svn_bin/visit-install more generic so that when a new site is added to resources/hosts that visit-install will pick it up. For instance, I added a custom launcher to the Utah hosts but forgot to up date visit-install so it would get copied over. "
2712,Pending,VisIt,Bug,Normal,vtkCleanPolydata,"","","",Allen Sanderson,11/07/2016,"",0,"",11/07/2016 01:44 pm,11/09/2016 12:39 pm,3 - Occasional,2 - Minor Irritation,trunk,"","",All,Any,"With the ability to draw curves as tubes and ribbons as well as end points via Pseudocolor plot curves must be ""clean"" (free of duplicate points). vtkCleanPolydata cleans in such a way that it removes duplicate points based on position only which can skew the results. Think about a critical point where the velocity is near zero. There is still advancement in time but little spatial displacement. vtkCleanPolydata would just remove the duplicate point but the time values are actually quite different. Thus it would appear that the advection stopped when it did not as the last points would be removed. 

Within the integral curve operator I implemented some smart cleaning so tubes and ribbons could be drawn correctly without skewing the result. Similar smart cleaning needs to be implemented in the Pseudocolor plot. Note: the smart cleaning in the integral curve operator is based on the velocity. "
2999,Resolved,VisIt,Bug,Normal,Viewing a dataset that has a single point in it causes Visit to hang,"","","",Allen Sanderson,01/24/2018,"",0,"",01/24/2018 06:37 pm,01/25/2018 05:11 pm,5 - Always,4 - Crash / Wrong Results,2.12.3,"","",All,Any,"I have a dataset with a single point in it. When it is rendered the viewing bounds are a point - ie. the min and the max are the same. This causes the issue. 

I have seen a similar issue when rendering a series of points but then doing a subsection that results in a single point. I have not been able to successfully replicate the issue (mostly as I was not sure of the cause and it was with the selection). But I can with the single point data file.  I have attached test dataset with one and two points.

Having a single point affects the view and the bounding box so I am not sure how to solve the issue.
"
